lib/platform/gitlab/index.js

```diff
@@ -381,12 +381,93 @@ function addReviewers(iid, reviewers) {
   logger.warn('Unimplemented in GitLab: approvals');
 }
 
-async function ensureComment() {
-  // Todo: implement. See GitHub API for example
+async function getComments(issueNo) {
+  // GET https://gitlab.com/api/v4/projects/:owner/:repo/merge_requests/:number/notes
+  logger.debug(`Getting comments for #${issueNo}`);
+  const url = `https://gitlab.com/api/v4/projects/${config.parentRepo ||
+    config.repository}/merge_requests/${issueNo}/notes`;
+  const comments = (await get(url, { paginate: true })).body;
+  logger.debug(`Found ${comments.length} comments`);
+  return comments;
+}
+
+async function addComment(issueNo, body) {
+  // POST https://gitlab.com/api/v4/projects/:owner/:repo/merge_requests/:number/notes
+  await get.post(
+    `https://gitlab.com/api/v4/projects/${config.parentRepo ||
+      config.repository}/merge_requests/${issueNo}/notes`,
+    {
+      body: { body },
+    }
+  );
+}
+
+async function editComment(issueNo, commentId, body) {
+  // PATCH https://gitlab.com/api/v4/projects/:owner/:repo/merge_requests/:number/notes/:id
+  await get.patch(
+    `https://gitlab.com/api/v4/projects/${config.parentRepo ||
+      config.repository}/merge_requests/${issueNo}/notes/${commentId}`,
+    {
+      body: { body },
+    }
+  );
+}
+
+async function deleteComment(issueNo, commentId) {
+  // DELETE https://gitlab.com/api/v4/projects/:owner/:repo/merge_requests/:number/notes/:id
+  await get.delete(
+    `https://gitlab.com/api/v4/projects/${config.parentRepo ||
+      config.repository}/merge_requests/${issueNo}/notes/${commentId}`
+  );
 }
 
-async function ensureCommentRemoval() {
-  // Todo: implement. See GitHub API for example
+async function ensureComment(issueNo, topic, content) {
+  const comments = await getComments(issueNo);
+  let body;
+  let commentId;
+  let commentNeedsUpdating;
+  if (topic) {
+    logger.debug(`Ensuring comment "${topic}" in #${issueNo}`);
+    body = `### ${topic}\n\n${content}`;
+    comments.forEach(comment => {
+      if (comment.body.startsWith(`### ${topic}\n\n`)) {
+        commentId = comment.id;
+        commentNeedsUpdating = comment.body !== body;
```

Comment:
Please check with a real gitlab repo to make sure that comments are not updated unnecessarily each run. e.g. make sure that GitLab doesn't add or remove newlines, for example. If there is any string difference then we will keep unnecessarily editing the comment every run.